tests/v2/validation/provisioning/providers.go

```diff
@@ -0,0 +1,59 @@
+package provisioning
+
+import (
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/cloudcredentials"
+	"github.com/rancher/rancher/tests/framework/extensions/cloudcredentials/aws"
+	"github.com/rancher/rancher/tests/framework/extensions/cloudcredentials/digitalocean"
+	"github.com/rancher/rancher/tests/framework/extensions/cloudcredentials/harvester"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type CloudCredFunc func(rancherClient *rancher.Client) (*cloudcredentials.CloudCredential, error)
+type MachinePoolFunc func(generatedPoolName, namespace string) *unstructured.Unstructured
+
+type Provider struct {
+	Name            string
+	MachineConfig   string
+	MachinePoolFunc MachinePoolFunc
+	CloudCredFunc   CloudCredFunc
+}
+
+func CreateProvider(name string) Provider {
```

Comment:
Add a comment above this function explaining what it's used for. In the format of 
`// CreateProvider ...  ` I want to keep all of our exported functions well documented.